src/Features/LanguageServer/Protocol/WellKnownLspServerKinds.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.LanguageServer;
+
+internal enum WellKnownLspServerKinds
+{
+    /// <summary>
+    /// Roslyn LSP server for razor c# requests.
+    /// </summary>
+    RazorLspServer,
+
+    /// <summary>
+    /// Roslyn LSP server for liveshare guests.
+    /// </summary>
+    LiveShareLspServer,
+
+    /// <summary>
+    /// Roslyn LSP server always activated in VS.
+    /// </summary>
+    AlwaysActiveLspServer,
+
+    /// <summary>
+    /// Roslyn LSP server for non-VS use cases.
+    /// </summary>
+    CSharpVisualBasicLspServer,
+
+    /// <summary>
+    /// XAML LSP servers.
+    /// </summary>
+    XamlLspServer,
+    XamlLspServerDisableUX,
```

Comment:
=-o

we have a xaml lsp server?

Reply:
We do in fact, I believe they access solution snapshots and so take advantage of our queueing mechanisms.

Reply:
@CyrusNajmabadi Yes, it's used for LiveShare and it's still internally experimental for the regular client.